### PR TITLE
feat(migration): MsTeams backfill default installation type

### PR DESCRIPTION
**Pull request by NisanthanNanthakumar**
## Objective:
Add installation_type field to the metadata column for existing msteams records in sentry_integration.

Requires https://github.com/getsentry/sentry/pull/36534 to be merged first.

### CODE REVIEW

**Review by ceorourke — APPROVED** [review]
Looks good to me, should wait until @getsentry/owners-migrations approves though 👍

File: src/sentry/migrations/0300_backfill_msteams_default_installation_type.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def backfill_msteams_installation_type(apps, schema_editor):
+    Integration = apps.get_model("sentry", "Integration")
+
+    for integration in RangeQuerySetWrapperWithProgressBar(
+        Integration.objects.filter(provider="msteams")
```
Comment (markstory) [inline src/sentry/migrations/0300_backfill_msteams_default_installation_type.py:13]: There are ~3k rows that match this criteria. You should be able to run this during the deploy window, but if you want to be safe you could make this migration `is_dangerous`.

**Review by markstory — APPROVED** [review]
Looks good to me.

**Review by wedamija — COMMENTED** [review]
Please write a test for this backfill, here's an example of how: https://github.com/getsentry/sentry/blob/52ca39e11b499f284f73376831c54ac8f837dc6b/tests/sentry/migrations/test_0297_backfill_alert_owners.py#L8

**Pull request merged**